File: src/main/java/com/example/webfluxdemo/adapters/TranslationAdapter.java
package com.example.webfluxdemo.adapters;

import com.example.webfluxdemo.AbstractAdapter;
import org.springframework.stereotype.Component;
import org.springframework.web.client.RestTemplate;

@Component
public class TranslationAdapter extends AbstractAdapter<String> {

    private static final String URL_PREFIX = "http://localhost:8040/api/translations/";

    public TranslationAdapter() {
        this(new RestTemplate());
    }

    public TranslationAdapter(RestTemplate restTemplate) {
        super(restTemplate, URL_PREFIX);
    }

    public String getTranslation(String word, String language) {
        return sendRequest(word + "/" + language, String.class);
    }

}

File: src/main/java/com/example/webfluxdemo/adapters/TranslationAdapter.kt
package com.example.webfluxdemo.adapters

import com.example.webfluxdemo.AbstractAdapter
import org.springframework.stereotype.Component
import org.springframework.web.reactive.function.client.WebClient
import reactor.core.publisher.Mono

private const val URL_PREFIX: String = "http://localhost:8040/api/translations/"

@Component
class TranslationAdapter(webClient: WebClient = WebClient.builder().build()) : AbstractAdapter(webClient, URL_PREFIX) {
    fun getTranslation(word: String, language: String): Mono<String> =
            sendRequest("$word/$language")
}